simulator/src/ledger/lookup.rs

```diff
@@ -0,0 +1,91 @@
+use radix_engine::engine::ScryptoInterpreter;
+use radix_engine::ledger::ReadableSubstateStore;
+use radix_engine::model::GlobalAddressSubstate;
+use radix_engine::types::{
+    GlobalAddress, GlobalOffset, NonFungibleIdType, RENodeId, ResourceAddress, ResourceManagerId,
+    ResourceManagerOffset, ResourceType, SubstateId, SubstateOffset,
+};
+use radix_engine::wasm::DefaultWasmEngine;
+use radix_engine_stores::rocks_db::RadixEngineDB;
+
+use crate::resim::get_data_dir;
+
+pub fn lookup_id_type(
+    resource_address: &ResourceAddress,
+) -> Result<NonFungibleIdType, LedgerLookupError> {
+    let scrypto_interpreter = ScryptoInterpreter::<DefaultWasmEngine>::default();
+    let substate_store = RadixEngineDB::with_bootstrap(
+        get_data_dir().map_err(|_| LedgerLookupError::FailedToGetLocalSubstateStorePath)?,
+        &scrypto_interpreter,
+    );
+
+    // Reading the global address substate to get the ResourceManagerId from there
+    let global_address = GlobalAddress::Resource(*resource_address);
+    let node_id = RENodeId::Global(global_address);
+    let offset = SubstateOffset::Global(GlobalOffset::Global);
+    let substate_id = SubstateId(node_id, offset);
+    let global_address_substate = substate_store.get_substate(&substate_id).map_or(
+        Err(LedgerLookupError::GlobalAddressNotFound(global_address)),
+        |value| Ok(value.substate.global().clone()),
+    )?;
+
+    let resource_manager_id = match global_address_substate {
+        GlobalAddressSubstate::Resource(id) => id,
+        _ => panic!(
+            "A global resource address can not point to anything other than a resource manager"
+        ),
+    };
+
+    // Reading the resource manager substate from the substate store
+    let node_id = RENodeId::ResourceManager(resource_manager_id);
+    let offset = SubstateOffset::ResourceManager(ResourceManagerOffset::ResourceManager);
+    let substate_id = SubstateId(node_id, offset);
+    let resource_manager_substate = substate_store.get_substate(&substate_id).map_or(
+        Err(LedgerLookupError::ResourceManagerNotFound(
```

Comment:
Should panic due to illegal state?

Reply:
Yes you're right. If the global address was found then the resource manager should be found as well. will change this to a panic.